modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/UpdateResult.scala

```diff
@@ -0,0 +1,27 @@
+package org.tessellation.dag.l1.infrastructure
+
+import cats.effect.Sync
+import cats.syntax.flatMap._
+
+import scala.util.control.NoStackTrace
+
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+
+sealed trait UpdateResult
+case object Success extends UpdateResult
+case class UnexpectedStateShift(message: String) extends NoStackTrace with UpdateResult {
```

Comment:
I also thought about adding one more case of non-fatal-unexpected-shift which would only log and return unit.